streaming/streaming.go

```diff
@@ -756,6 +759,26 @@ func CreateStreamingConfig(pl profiles.DataShard) StreamingCfg {
 	return streamingCfg
 }
 
+// Returns a string that encodes the project number like `projects/12345`
+func GetProjectNumber(ctx context.Context, projectID string) string {
```

Comment:
Is it better to return just the number instead of `projects/12345` for the sake of consistency? I think probably returning the `projects/12345` is better if we did it for `locations/%s` and `streams/%s`, but doing it only for project the URL confusing to read - 

Now
```
fmt.Sprintf("%s/locations/%s/streams/%s", projectNumber, datastreamCfg.StreamLocation, datastreamCfg.StreamId)
```

vs

Earlier
```
fmt.Sprintf("projects/%s/locations/%s/streams/%s", projectID, datastreamCfg.StreamLocation, datastreamCfg.StreamId)
```

Reply:
I had this in an earlier version :), but changed it.
There are two reasons why this gets it a bit clumsy:
1. The API returns "projects/<Number>" which we will have to strip and re-join at multiple places. 
2. If we just return the number, the type would still be a string, and conversion up and down would be another un-necessary operation if we go down that path.

I also agree with the point that the location and stream is plumbed when the API is called.
I will make the variable name `ProjectNumberResource` to avoid confusion. The convention for any resource in Google APIs is `ResourceType/ResoucrceId` here the ID being the actual project number.